src/utils/auth.ts

```diff
@@ -176,27 +177,32 @@ export default class Auth {
           },
           (error) => {
             console.error(error)
-            this.logout()
-            reject(error)
+            this.logout().then(() => reject(error))
           },
         )
       }
     })
   }
 
   clearLocalStorage() {
```

Comment:
What about calling this `clearSessionData` since it is wiping out session data from both a cookie and local storage?

Reply:
I'm all for improvements, but would probably defer them for now. I think this needs to be deeply reconsidered and probably deserves some xstate.